addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -0,0 +1,352 @@
+                'GrossAmount': gross_amount,
+                'TaxesOutputs': taxes_output,
+                'TaxesWithheld': tax_withheld_output,
+            })
+            items.append(invoice_line_values)
+            taxes += taxes_output
+            taxes_withheld += tax_withheld_output
+        return items, taxes, taxes_withheld, totals
+
+    @api.model
+    def _l10n_es_edi_facturae_export_facturae(self):
+        """
+        Produce the Facturae XML data for the invoice.
+        :return: (data needed to render the full template, data needed to render the signature template)
+        """
+        self.ensure_one()
+        company = self.company_id
+        partner = self.commercial_partner_id
+
+        if not company.vat:
+            raise UserError(_('The company needs a set tax identification number or VAT number'))
+
+        if not partner.vat:
+            raise UserError(_('The partner needs a set tax identification number or VAT number'))
+
+        eur_curr = self.env['res.currency'].search([('name', '=', 'EUR')])
+        inv_curr = self.currency_id
+        legal_literals = self.narration.striptags() if self.narration else False
+        legal_literals = legal_literals.split(";") if legal_literals else False
+        partner_name = {'firstname': 'UNKNOWN', 'surname': 'UNKNOWN', 'surname2': ''}
+        if not partner.is_company:
+            name_split = [part for part in partner.name.replace(', ', ' ').split(' ') if part]
+            if len(name_split) > 2:
+                partner_name['firstname'] = ' '.join(name_split[:-2])
+                partner_name['surname'], partner_name['surname2'] = name_split[-2:]
+            elif len(name_split) == 2:
+                partner_name['firstname'] = ' '.join(name_split[:-1])
+                partner_name['surname'] = name_split[-1]
+
+        template_values = {
+            'Modality': '',
+            'ThirdParty': False,
+            'BatchIdentifier': '',
+            'InvoicesCount': '',
+            'TotalInvoicesAmount': {'TotalAmount': '', 'EquivalentInEuros': False, },
+            'TotalOutstandingAmount': {'TotalAmount': '', 'EquivalentInEuros': False, },
+            'TotalExecutableAmount': {'TotalAmount': '', 'EquivalentInEuros': False, },
+            'InvoiceCurrencyCode': '',
+            'Invoices': [],
+        }
+        self_party_values = {
+            'PartyIdentification': False,
+            'AdministrativeCentres': [],
+        }
+        partner_party_values = deepcopy(self_party_values.copy())
+        signature_values = {'SigningTime': '', 'SignerRole': '', }
+
+        if self.move_type == "entry":
+            return False
+
+        invoice_issuer_signature_type = 'supplier' if self.move_type == 'out_invoice' else 'customer'
+
+        need_conv = bool(inv_curr != eur_curr)
+
+        conversion_rate = self.amount_total_in_currency_signed / self.amount_total_signed if self.amount_total_signed else 0.
```

Comment:
```suggestion
        conversion_rate = abs(self.amount_total_in_currency_signed / self.amount_total_signed) if self.amount_total_signed else 0.
```

Reply:
I will do, but if both sides are signed, then it should always be positive right?